### DISCUSSION TITLE
How much RAM is needed for the RPi device?

### DISCUSSION THREAD

**Discussion by ritog** [microsoft/IoT-For-Beginners#343] [Q&A] [answered]
I have searched through the repo, and I could not find the amount of memory necessary in the Raspberry Pi device.

I am planning to buy an RPi 4, but unsure about the memory amount.

Which one should I buy- 2GB, 4GB, or 8GB?

Is 2 gigs of RAM enough for this course? Or should I buy another one?

**Comment by jimbobbennett** [comment, marked as answer]
I've just added a note to the hardware.md file to clarify this.

If you are planning to run VS Code directly on the Pi, then any Pi 4 with 2GB of RAM or more will work. The new 1GB on doesn't have enough RAM to run VS Code and the Python language server well on the device.

If you are planning to run VS Code on your PC or Mac and connect using remote SSH, then any Pi 4 will work fine, including the new 1GB version.
